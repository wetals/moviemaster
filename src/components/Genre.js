import React from "react";

const Genre = props => {
  return (
    <div className="genre__tags">
      {props.genre_ids.map(genre => (
        <span key={genre} className="genre">
          {props.genresList[genre]}
        </span>
      ))}
    </div>
  );
};

export default Genre;
