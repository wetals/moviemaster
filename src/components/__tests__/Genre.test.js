import React from "react";
import { shallow } from "enzyme";
import renderer from "react-test-renderer";

import Genre from "../Genre";

const testData = {
  genre_ids: [1, 2],
  genresList: {
    1: "foo",
    2: "bar"
  }
};

describe("Genre component", () => {
  it("renders properly", () => {
    const wrapper = shallow(<Genre {...testData} />);
    const element = wrapper.find("span");
    expect(element.length).toBe(testData.genre_ids.length);
    expect(element.first().text()).toBe("foo");
    expect(element.last().text()).toBe("bar");
  });

  it("renders snapshot properly", () => {
    const tree = renderer.create(<Genre {...testData} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
