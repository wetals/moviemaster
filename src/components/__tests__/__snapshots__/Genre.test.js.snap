// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Genre component renders snapshot properly 1`] = `
<div
  className="genre__tags"
>
  <span
    className="genre"
  >
    foo
  </span>
  <span
    className="genre"
  >
    bar
  </span>
</div>
`;
